src/tests/ftest/io/nvme_object.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+import threading
+import avocado
+
+from daos_api import DaosApiError
+from test_utils import TestPool, TestContainer
+from apricot import TestWithServers
+
+
+def container_write(container, record):
+    """Method to write to a container
+
+       Args:
+         container: instance of TestContainer
+         record: record size to be written
+    """
+    # update record qty
+    container.record_qty.update(record)
+    print("\nRecord Size:{}".format(container.record_qty))
+
+    # write multiple objects
+    container.write_objects()
+
+def container_read(container):
+    """Method to read and verify the written data
+
+       Args:
+         container: instance of TestContainer
+    """
+
+    # read written objects and verify
+    container.read_objects()
+
+def test_runner(self, size, record_size):
```

Comment:
What do you think about re-working this method a bit into one that populates a `self.jobs = {"write": [], "read": []}` dict of lists using a list of pool sizes to determine how many pools and containers to create and a list of record sizes to determine how many `container_write` and `container_read` threads to create?  It would still manage the threads after they are created, but it I think it would simplify `test_nvme_object_multiple_pools` a bit.